lib/poker_hands/hand_rankings/flush.rb

```diff
@@ -1,5 +1,9 @@
 class Flush
   def check(hand)
-    true
+    return false if hand.any? { |card| card.suit != hand.first.suit}
+
+    hand.sort_by! { |card| card.rank.to_i }
+
+    !hand.each_cons(2).all? { |card, next_card| next_card.rank.to_i == card.rank.to_i + 1 }
```

Comment:
Is this what a flush is?

Reply:
Yes, all cards of the same suit, which we check on line 3, and not in a sequence, which we check on line 7